custom_components/solarman/inverter_definitions/kstar_hybrid.yaml

```diff
@@ -4,28 +4,24 @@
 #
 # This inverter exposes its data in the following registers (although not all of them are used by this definition file):
 #
-#  - INPUT_REGISTERS   3000 - 3660 decimal, 0x0BB8 - 0x0E4C hexadecimal
-#  - HOLDING_REGISTERS 3200 - 3237 decimal. 0x0C80 - 0x0C9B  hexadecimal
-#  - INPUT_REGISTERS   3292 - xxxx decimal. Not even sure about the start, could be even lower
-#
-# Each request can get a maximum of 125 registers as per modbus protocol (start and end included), so we need to
-# split up the list of used registers into multiple requests of maximum 125 registers each.
```

Comment:
Hi @davidrapan 

I originally had to split the requests in subsets of 125 registers, because if there were more, it failed fetching. When I investigated time ago, what I found was that the protocol only allows up to 125 registers per request.

I can't find this splitting in you code, is this now handled by the pysolarman library, or did I failed to find it?

Reply:
It's controlled by the min_span in the default section of the definition file.

25 is the default value which works for most of the devices. If you have problems you can set it to lower value.

Reply:
Ah, I wondered what that `min_span` was for. Shouldn't it be named `max_span`, as that is the maximum span of registers that can t requested?

Also, why a so small value? The maximum value is 125, so 25 means 5 times more requests for contiguous registers.

Reply:
Cause it means something little bit different. It means what minimum space between two consecutive requested registers shoud be included in single request.

For example let's say I want to request two registers:
- If they are 500 and 520 they will be included in single request
- If they are 500 and 230 it will generate two separated requests

So basically it does control how much registers can be in single request indirectly.

There still could be hardcoded that max size of 125 but current constraints were sufficient enough so far.

Reply:
I see.

Yes, I was checking the registers I need to query for this inverter, and it seems that with a min_span of  25 it needs to make 6 requests.

```txt
  3000 3001
  3012 3013
  3024 3025
  3036 3037 3038 3039 3040 3041 3042
  3044 3045 3046 3047 3048 3049 3050 3051 3052 3053 3054 3055 3056 3057
  3062 3063 3064 3065 3066 3067 3068

  3097 3098 3099 3100
  3109 3110 3111 3112 3113 3114 3115 3116 3117 3118 
  3119 3120 3121 3122 3123 3124 3125 3126
  3135 3136 3137
  3144
  3147 3148 3149 3150 3151 3152 3153

  3193

  # 0x3
  3200 3201 3202 3203 3204 3205 3206 3207 3208 3209 
  3210 3211 3212 3213 3214 3215 3216 3217 

  3228 3229 3230 3231 3232 3233 3234 3235 3236 3237 3238

  3292 3293 3294
  3299 3300 3301
```

However, if I needed to get the register 3075 (BAT Health), it would mean that it would try to fetch from 3000 to 3153, and would fail, forcing me to reduce the `min_span` and make more requests.

I suppose splitting by maximum consecutive registers up a max of 125 would make it more resilient, reduce the number of requests and make `min_span` useless (although harder to program). In my case, it would make only 4 requests from the original 6:

```txt
300-3124
3125-3249
3200-3217 (0x3)
3250-3301
```

If I enable `requests_fine_control` to requests only these 4 sets of registers, would `min_span` still kick in and split them into more requests?

Reply:
No requests_fine_control will override it completely and use exactly the sets which are in the requests section.

You know I can try to set there that 125 hard limit.

Reply:
[feat: Hardcoded max modbus request size](https://github.com/davidrapan/ha-solarman/commit/c51ff0b4dc29f11797d90c732fdd56b140dad9b5)

Reply:
With that commit you can set min_span to 55 to get the lowest number of request possible. It's 4 requests.

Reply:
> You know there is the other thing that not all registers has to be requested every time. There are some which are. But for example some static one could have update_interval: 300 to be requested every 5m. I believe thats for example exactly your 3200 - 3217.

True, I didn't think about different update intervals. In this PR I have some registers to update less frequently, as they should not change frequently, such us the inverter information (model, version, SN...).

> With that commit you can set min_span to 55 to get the lowest number of requests possible. It's 4 requests.

Thanks, I'll try.

I haven't updated to latest version due to the sensor renaming, I'm waiting until I have some time to sit and rename all changed entities.

Reply:
> I haven't updated to latest version due to the sensor renaming, I'm waiting until I have some time to sit and rename all changed entities.

Sure thing! If you'd had any problems and need any help, don't hesitate to reach out to me! 😉